lib/rubocop/cop/style/lambda.rb

```diff
@@ -25,19 +27,19 @@ def on_block(node)
           length = lambda_length(node)
 
           if selector != '->' && length == 1
-            add_offense_for_single_line(block_method, args)
+            add_offense_for_single_line(node, block_method.loc.expression, args)
           elsif selector == '->' && length > 1
-            add_offense(block_method, :expression, MULTI_MSG)
+            add_offense(node, block_method.loc.expression, MULTI_MSG)
           end
         end
 
         private
 
-        def add_offense_for_single_line(block_method, args)
+        def add_offense_for_single_line(block_node, expression, args)
```

Comment:
Instead of `expression`, call the parameter `location`. It's used for selecting what will be highlighted in the report, and in the base class it's called `loc`.